### ISSUE TITLE
Carousel with two images slide transitions the wrong way at edge.

### ISSUE THREAD

**Issue by klapeyre** [reactstrap/reactstrap#1341]
- components: `Carousel`
- reactstrap version `#6.5.0`
- import method `csj`
- react version `#16.6.3`
- bootstrap version `#4.1.3`

### What is happening?

An image carousel with only two images will not slide transition from slide at index 1 to slide at index 0 in the correct direction. Instead, it slides back (instead of back around, if that makes sense?). Clicking the carets is still semi-user friendly, but if the carousel is swipeable, then the user interaction makes no sense.

### What should be happening?

Move from index 0 to index 1, with slide transition right to left. Then again in the same direction.

### Steps to reproduce issue

1. Setup simple carousel as seen here: https://reactstrap.github.io/components/carousel/
2. Remove one of the items so there are only two.

### Error message in console

```
No error is seen, the logic is carried out, but it may be incorrect.
```

### Code

Second block in if statement, if activeIndex is 1 and nextProps.activeIndex is 0, it will swipe left.

```
componentWillReceiveProps(nextProps) {
    this.setInterval(nextProps);
    // Calculate the direction to turn
    if (this.props.activeIndex + 1 === nextProps.activeIndex) {
      this.setState({ direction: 'right' });
    } else if (this.props.activeIndex - 1 === nextProps.activeIndex) {
      this.setState({ direction: 'left' });
    } else if (this.props.activeIndex > nextProps.activeIndex) {
      this.setState({ direction: this.state.indicatorClicked ? 'left' : 'right' });
    } else if (this.props.activeIndex !== nextProps.activeIndex) {
      this.setState({ direction: this.state.indicatorClicked ? 'right' : 'left' });
    }
    this.setState({ indicatorClicked: false });
  }
```

**Comment by cbaxter713** [comment]
@klapeyre did you ever find a solution to this issue?  I am up against the same problem

**Comment by klapeyre** [comment]
I think I ended up patching it client side, guarding for the case where there are two images in the carousel.

**Comment by cbaxter713** [comment]
@klapeyre yeah I am trying to do the same thing, but haven't come up with a successful patch yet.  Mind sharing any tips on what you did?

**Comment by klapeyre** [comment]
With the example they provided in the docs here (https://reactstrap.github.io/components/carousel/) - I added a check in the next() and previous() function implementations. If your activeIndex is 1 and array length is 2 (this confirms your carousel has two images and you are at the end of it), I set the index to 2 and then immediately after that set the index to 0. This gives the visual impression that it is still swiping right. Do the same for the implementation of previous().
